### DISCUSSION TITLE
Error: Failed to launch the browser process!

### DISCUSSION THREAD

**Discussion by adityadees** [spatie/browsershot#800] [Bugs]
I just get the error 

Error Output:
================
[1m[43m[30m
  Puppeteer old Headless deprecation warning:[0m[33m
    In the near future `headless: true` will default to the new Headless mode
    for Chrome instead of the old Headless implementation. For more
    information, please see https://developer.chrome.com/articles/new-headless/.
    Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`
    If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.[0m

Error: Failed to launch the browser process!
chrome_crashpad_handler: --database is required
Try 'chrome_crashpad_handler --help' for more information.
[2480:2480:1213/145419.328318:ERROR:socket.cc(120)] recvmsg: Connection reset by peer (104)

TROUBLESHOOTING: https://pptr.dev/troubleshooting

    at Interface.onClose (/usr/local/lib/node_modules/puppeteer/node_modules/@puppeteer/browsers/lib/cjs/launch.js:271:24)
    at Interface.emit (node:events:529:35)
    at Interface.close (node:internal/readline/interface:534:10)
    at Socket.onend (node:internal/readline/interface:260:10)
    at Socket.emit (node:events:529:35)
    at endReadableNT (node:internal/streams/readable:1368:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

anyone know how to fix that?

**Comment by RealMrHex** [comment]
Same error!
Is there any solution?

**Comment by aldinokemal** [comment]
same with me running on docker with centos 7 server

**Reply by RealMrHex** [reply]
@aldinokemal Same here and still not solved :(

**Reply by aldinokemal** [reply]
I have fixed by changing my browser to chrome headless

now is working in my server

**Comment by matticustard** [comment]
I have a VPS (AlmaLinux) with multiple accounts and domains. I was really struggling with this "Failed to launch the browser process! chrome_crashpad_handler: --database is required" issue over the past day after having performed some general updates. This was never an issue previously.

While there may be various causes, what I finally discovered in my case was that multiple user accounts were fighting over a single Chromium `.config` directory. Chromium was putting this folder in the system temp folder (`/tmp/.config`). This allowed a single user (the one who established and owned the folder) to generate Browsershot documents, but no others — or they would get the "Failed to launch the browser process! chrome_crashpad_handler: --database is required" issue. The solution was to assign a custom `.config` directory for each user. This is NOT the same as the user-data directory, of which changing had no effect. Now, all users can generate documents without colliding.

```
->setEnvironmentOptions([
    'CHROME_CONFIG_HOME' => '/home/'.get_current_user().'/.config'
])
```

Hopefully, this may help someone else. I was able to discover many instances of the issue online, but no other solutions worked for me.

**Reply by holymp2006** [reply]
This solution worked for me. However, I had to tweak it slightly due to another "Permission denied" error that I received.
```
       ->setEnvironmentOptions([
                'CHROME_CONFIG_HOME' => storage_path('app/chrome/.config')
            ])
```

**Reply by hzeeshan** [reply]
finally, for me this solution worked after many hours of struggle. In my case the app is running inside docker container.

```
$pdf = Browsershot::html($html)
    ->setChromePath("/usr/bin/chromium")
    ->noSandbox()
    ->waitUntilNetworkIdle()
    ->ignoreHttpsErrors()
    ->setEnvironmentOptions([
        'CHROME_CONFIG_HOME' => storage_path('app/chrome/.config')
    ])
```

**Reply by crivion** [reply]
DAAAARNNN! THANK YOUUUUUUUUU!

> finally, for me this solution worked after many hours of struggle. In my case the app is running inside docker container.
> 
> ```
> $pdf = Browsershot::html($html)
>     ->setChromePath("/usr/bin/chromium")
>     ->noSandbox()
>     ->waitUntilNetworkIdle()
>     ->ignoreHttpsErrors()
>     ->setEnvironmentOptions([
>         'CHROME_CONFIG_HOME' => storage_path('app/chrome/.config')
>     ])
> ```

**Comment by kfina-planxy** [comment]
This fix worked for us:
create a `.puppeteerrc.cjs` config file in the root directory of your laravel application:
```
const {join} = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
    // Changes the cache location for Puppeteer.
    cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};
```
This circumvents user permission issues, where your webserver might not have access to chrome in other locations.

And also append these parameters to your browsershot:
```
$browsershot->addChromiumArguments([
    'database' => 'crashpad_database'
])->setChromePath(base_path('.cache/puppeteer/chrome-headless-shell/linux-133.0.6943.126/chrome-headless-shell-linux64/chrome-headless-shell'));
```
This ensures that chrome headless is used instead of base chrome and the database is found.
(If you still get the error, create a "crashpad_database" folder in the headless chrome directory: ".cache/puppeteer/chrome-headless-shell/linux-133.0.6943.126/chrome-headless-shell-linux64/chrome-headless-shell")
